pangeo_forge_orchestrator/catalog.py

```diff
@@ -0,0 +1,174 @@
+import json
+from pathlib import Path
+
+import shapely.geometry
+import xarray as xr
+import xstac
+from rich import print
+
+from .components import Bakery, FeedstockMetadata
+
+
+def generate(
+    bakery_name,
+    run_id,
+    bakery_database_path=None,
+    bakery_stac_relative_path=None,
+    feedstock_metadata_url_base=None,
+    print_result=False,
+    to_file=False,
+    endpoints=["s3", "https"],
+):
+    write_access = True if to_file else False
+    item_result, feedstock_id, bakery = _generate(
+        bakery_name=bakery_name,
+        run_id=run_id,
+        bakery_database_path=bakery_database_path,
+        bakery_stac_relative_path=bakery_stac_relative_path,
+        feedstock_metadata_url_base=feedstock_metadata_url_base,
+        endpoints=endpoints,
+        write_access=write_access,
+    )
+    if print_result:
+        print(item_result)
+    if not to_file:
+        return item_result
+    elif to_file:
```

Comment:
No need to have an `if..elif..else` if you return early. You can remove this indentation.